### PR TITLE
codeblocks: 13.12 -> 16.01

### PR DESCRIPTION
**Pull request by ndowens**
###### Motivation for this change
Update

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/editors/codeblocks/default.nix [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
   src = fetchurl {
-    url = "mirror://sourceforge/codeblocks/Sources/${version}/codeblocks_${version}-1.tar.gz";
-    sha256 = "044njhps4cm1ijfdyr5f9wjyd0vblhrz9b4603ma52wcdq25093p";
+    url = "mirror://sourceforge/codeblocks/Sources/${version}/codeblocks_${version}.tar.gz";
+    sha256 = "00sskm91r20ywydwqwx6v7z3nwn9lyh5297c5wp3razldlh9vyrh";
   };
 
-  buildInputs = [ automake autoconf libtool pkgconfig file zip wxGTK gtk2 ]
+  nativeBuildInputs = [ autoreconfHook pkgconfig ];
+  buildInputs = [ libtool file zip wxGTK gtk2 ]
```
Comment (7c6f434c) [inline pkgs/applications/editors/codeblocks/default.nix]: Are `zip` and `libtool` (and `file`) used by the installed application?

Intuitively I feel like they should be build-time-only dependencies used by the build system on the host (so, `nativeBuildInputs`), but I may be wrong.

**Comment by ndowens** [comment]
Wasn't sure if libtool was a build only dep or needed by run

**Comment by 7c6f434c** [comment]
Checked, not in the closure.

**Comment by 7c6f434c** [comment]
(Pushed the updated version)

File: pkgs/applications/editors/codeblocks/default.nix (hunk 1 above)
Comment (vcunat) [inline pkgs/applications/editors/codeblocks/default.nix]: Beware in general, as e.g. some packages link against `${libtool.lib}/lib/libltld.so`.

**Comment by 7c6f434c** [comment]
>Beware in general, as e.g. some packages link against `${libtool.lib}/lib/libltld.so`.

I really hope they don't do it in a complicated static way (there is no
run-time dependency, at least)

**Comment by vcunat** [comment]
Well, *if* someone attempted a real cross-compilation and we had this incorrectly, the linking should fail IIRC, whether it's static or dynamic (stuff compiled for different targets).

**Comment by 7c6f434c** [comment]
Well, it was obviously never attempted before, and linking `libltld.so`
statically should be an interesting exercise in build systems anyway.

(I do know that arch mismatch would kill static linking)

**Pull request closed without merging**